### PR TITLE
Fix incremental build restore: move generated test_runtime project.json to bin

### PR DESCRIPTION
**Pull request by dagood**
In some cases incremental builds didn't work because package restore fails on the generated test_runtime project.json. From my email:

> That project.json is generated by `tests\runtest.proj` to contain a dependency on the currently built CoreCLR package, and `runtest.proj` also restores it, using the local package output dir as the package source. Unfortunately, the generated project.json is left in the `tests\src\` tree, so it’s picked up by any recursive restores in later builds. If the packages dir is cleared and that generated project.json is still there, the restore will try to find the package on MyGet, and it will either hit that error or find a “matching” `-00` version.

The restore error is "`Unable to resolve 'Microsoft.NETCore.Runtime.CoreCLR (>= 1.0.2-rc3-24103-0)`"

This change moves the generated project.json into `bin/tests/test_runtime`. I've only done a tiny amount of local testing because the test build is going extremely slow for me. This PR will not fix existing repros, but should prevent this issue from reproing again in the future.

/cc @BruceForstall @geoffkizer @rahku @janvorli

### CODE REVIEW

**Comment by dagood** [comment]
@dotnet-bot test OSX x64 Checked Build and Test

**Comment by rahku** [comment]
This json file is used during helix test runs and not during local test runs. You would also need to make corresponding changes in tests\helix.targets & tests\helixperftasks.targets.
In helix.targets at https://github.com/dotnet/coreclr/blob/master/tests/helix.targets#L27 you should also add this project.lock.json to TestNugetProjectLockFile Itemgroup. in tests\helixperftasks.targets make changes here https://github.com/dotnet/coreclr/blob/master/tests/helixperftasks.targets#L17

**Comment by rahku** [comment]
CC: @dsgouda

**Comment by dsgouda** [comment]
Agree to what @rahku mentioned. Please make sure you can successfully run tests (perf and func) via helix with your changes. Let me know if I can help with it.

**Comment by rahku** [comment]
functional helix runs are currently broken so don't bother about running that currently. Just make the change I suggested in tests\helix.targets

**Comment by dagood** [comment]
Thanks for the pointers, I updated the helix targets. I haven't kicked off helix runs before, so once that's ready I'll need help to get that going. Do we have any docs on this? (I couldn't find any, but I don't really know where to look.)

File: tests/helixperftasks.targets [hunk 1]
```diff
@@ -15,8 +15,8 @@
     </GetPerfTestAssemblies>
 
     <PropertyGroup>
-      <TestRuntimeDependenciesJson>$(SourceDir)Common\test_runtime\project.json</TestRuntimeDependenciesJson>
-      <TestRuntimeProjectLockJson>$(SourceDir)Common\test_runtime\project.lock.json</TestRuntimeProjectLockJson>
+      <TestRuntimeDependenciesJson>$(TestRuntimeProjectJson)</TestRuntimeDependenciesJson>
```
Comment (dsgouda) [inline tests/helixperftasks.targets]: nit: I think we can simply replace all references to TestRuntimeDependenciesJson and TestRuntimeDependenciesJson with $(TestRuntimeProjectJson) and $(TestRuntimeProjectLockJson) respectively for this particular file.

Comment (dagood) [inline tests/helixperftasks.targets]: Good call, changed and removed the property group.

**Comment by dagood** [comment]
I kicked off a helix perf run (thanks for the help @dsgouda). After 190 tests it's at 100% pass rate and still running.

**Comment by dsgouda** [comment]
LGTM

**Comment by janvorli** [comment]
@dotnet-bot test OSX x64 Checked Build and Test please

**Comment by janvorli** [comment]
LGTM

**Pull request merged**